### PR TITLE
feat(cat-voices): registration confirm exit dialog

### PR DESCRIPTION
**Pull request by damian-molinski**
# Description

This PR adds confirmation `VoicesQuestionDialog` before closing `RegistrationDialog` when called via `Navigator.of(context).maybePop()`. `Navigator.of(context).pop()` will close dialog without question.

## Screenshots

Screenshot 2024-10-04 at 11 10 42

## Related Pull Requests

Part of #815 

## Please confirm the following checks

* [x] My code follows the style guidelines of this project
* [x] I have performed a self-review of my code
* [x] I have commented my code, particularly in hard-to-understand areas
* [x] I have made corresponding changes to the documentation
* [x] My changes generate no new warnings
* [x] I have added tests that prove my fix is effective or that my feature works
* [x] New and existing unit tests pass locally with my changes
* [x] Any dependent changes have been merged and published in downstream module

### CODE REVIEW

File: catalyst_voices/lib/widgets/modals/voices_question_dialog.dart [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+    return VoicesAlertDialog(
+      title: title,
+      icon: icon,
+      subtitle: subtitle,
+      content: content,
+      buttons: actions.map((e) => buildItem(context, item: e)).toList(),
+    );
+  }
+
+  Widget buildItem(
```
Comment (dtscalac) [inline catalyst_voices/lib/widgets/modals/voices_question_dialog.dart:92]: Consider to make it private

Comment (damian-molinski) [reply catalyst_voices/lib/widgets/modals/voices_question_dialog.dart]: Makes sense. Fixed

**Review by dtscalac — APPROVED** [review]
LGTM

**Comment by damian-molinski** [comment]
@digitalheartxs have a look pls

**Review by digitalheartxs — APPROVED** [review]
Looks good 🙌

**Pull request merged**